kratos/modeler/cad_tessellation_modeler.cpp

```diff
@@ -0,0 +1,352 @@
+            const auto r_brep_surface_geom = dynamic_cast<const BrepSurfaceType&>(r_aux_geometry);
+
+            std::vector<array_1d<double, 2>> boundary_loop_uv;
+            IndexType point_id = 0;
+
+            while (r_brep_surface_geom.HasGeometryPart(trim_index)) {
+
+                auto p_surface_geometry = cad_model_part.pGetGeometry(r_brep_surface_geom.Id());
+                auto p_surface_trim = p_surface_geometry->pGetGeometryPart(trim_index);
+                auto p_brep_curve_on_surface =
+                    dynamic_pointer_cast<BrepCurveOnSurfaceType>(p_surface_trim);
+
+                auto boundary_segment_tessellation = this->ComputeBoundaryTessellation(*p_brep_curve_on_surface);
+
+                boundary_loop_uv.resize(boundary_loop_uv.size() + boundary_segment_tessellation.size(), ZeroVector(2));
+
+                for (IndexType i = 0; i < boundary_segment_tessellation.size(); ++i) {
+
+                    boundary_loop_uv[point_id][0] = boundary_segment_tessellation[i][0];
+                    boundary_loop_uv[point_id][1] = boundary_segment_tessellation[i][1];
+
+                    point_id++;
+                }
+                trim_index++;
+            }
+
+            auto triangulation_uv = this->ComputeSurfaceTriangulation(
+                r_brep_surface_geom,
+                boundary_loop_uv);
+
+            array_1d<double,3> result = ZeroVector(3);
+            array_1d<double,3> local_coordinate = ZeroVector(3);
+
+            for (IndexType tri_i = 0; tri_i < triangulation_uv.size(); ++tri_i) {
+                for (IndexType point_i = 0; point_i < 3; ++point_i) {
+
+                    local_coordinate[0] = triangulation_uv[tri_i](point_i, 0);
+                    local_coordinate[1] = triangulation_uv[tri_i](point_i, 1);
+                    auto point_xyz = r_brep_surface_geom.GlobalCoordinates(result, local_coordinate);
+
+                    skin_model_part.CreateNewNode(
+                        node_id++,
+                        point_xyz[0],
+                        point_xyz[1],
+                        point_xyz[2]);
+                }
+            }
+
+            Properties::Pointer p_properties(new Properties(0));
+
+            // create elements in skin_model_part
+            for (IndexType element_i = 0; element_i < triangulation_uv.size(); ++element_i)
+            {
+                skin_model_part.CreateNewElement("Element3D3N",
+                    element_id++,
+                    {{vertex_id, vertex_id + 1, vertex_id + 2}},
+                    p_properties);
+
+                vertex_id += 3;
+            }
+        }
+    }
+}
+
+///@}
+///@name Private Operations
+///@{
+
+std::vector<array_1d<double, 2>> CadTessellationModeler::ComputeBoundaryTessellation(const BrepCurveOnSurfaceType& rBoundarySegment)
+{
+    KRATOS_ERROR_IF_NOT(mParameters.Has("absolute_chordal_error"))
+        << "Missing \"absolute_chordal_error\" in CadTessellationModeler Parameters" << std::endl;
```

Comment:
Rather than throwing an error, I'd provide a default value.

In order to do this, I'd create a `GetDefaultParameters` as we did for the  `Processes` in the base `Modeler` class.

Reply:
If you agree, we can do this in a standalone fast PR.

Reply:
I totally agree. We can create a PR for defaultparameters later on